include/ofi_sm2.h

```diff
@@ -0,0 +1,351 @@
+/*
```

Comment:
If we're moving the util_shm code inside the provider then we can move the ofi_smr.h code into smr.h (or a separate header). It was only in the common includes because it was in the util folder

Reply:
While we're here maybe also rename the file `util_sm2.c` to `sm2_util.c` for a consistent naming scheme.

Reply:
^also that!

Reply:
done